### PR TITLE
rbd: improve error checking

### PR DESCRIPTION
**Pull request by nixpanic**
Make it possible to easily validate the *Image and *Snapshot objects
before using them. This makes the use of go-ceph for RBD functions more
stable, and it is less likely that incorrect use causes Go panics.

### CODE REVIEW

**Comment by nixpanic** [comment]
@ansiwen could you review this please?

File: rbd/rbd.go [hunk 1]
```diff
@@ -45,12 +45,23 @@ const (
 	RbdFeaturesSingleClient = C.RBD_FEATURES_SINGLE_CLIENT
 )
 
+// bits for Image.validate() and Snapshot.validate()
+const (
+	imageNeedsName uint32 = 1 << iota
+	imageNeedsIOContext
```
[the remark this answers is not in the document]
Comment (ansiwen) [reply rbd/rbd.go]: And if there is something missing here, then there is also a test missing, right? ;-)

[the remark this answers is not in the document]
Comment (nixpanic) [reply rbd/rbd.go]: Yeah, Golang does some magic bit shifting like that!

Lots of tests are missing, I am adding those through #124

File: rbd/rbd.go [hunk 2]
```diff
@@ -117,6 +128,38 @@ func split(buf []byte) (values []string) {
 	return values
 }
 
+// test if a bit is set in the given value
+func hasBit(value, bit uint32) bool {
+	return (value & bit) == bit
```
[the remark this answers is not in the document]
Comment (nixpanic) [reply rbd/rbd.go]: Probably just habbit. And this works for multiple bits as well, in case `bit` has more set.

**Pull request merged**